src/Filecoin/Address.cpp

```diff
@@ -0,0 +1,196 @@
+// Copyright © 2017-2019 Trust.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Address.h"
+
+#include "../Base32.h"
+
+#include <TrezorCrypto/blake2b.h>
+
+using namespace TW::Filecoin;
+
+static const char BASE32_ALPHABET_FILECOIN[] = "abcdefghijklmnopqrstuvwxyz234567";
+static constexpr size_t checksum_size = 4;
+
+// Calculate the checksum of a byte slice.
+static std::array<uint8_t, checksum_size> calc_checksum(const std::vector<uint8_t>& bytes) {
```

Comment:
```suggestion
static std::array<uint8_t, checksum_size> calc_checksum(const Data& bytes) {
```
we prefer to always use `Data` alias now